autoload/lsp/utils/diff.vim

```diff
@@ -99,17 +99,23 @@ function! s:Length(lines, start_line, start_char, end_line, end_char)
   if l:adj_end_line >= len(a:lines)
     let l:adj_end_char = a:end_char - 1
   else
-    let l:adj_end_char = strlen(a:lines[l:adj_end_line]) + a:end_char
+    let l:adj_end_char = s:count_utf16_code_units(a:lines[l:adj_end_line]) + a:end_char
   endif
   if a:start_line == l:adj_end_line
     return l:adj_end_char - a:start_char + 1
   endif
-  let l:result = strlen(a:lines[a:start_line]) - a:start_char + 1
+  let l:result = s:count_utf16_code_units(a:lines[a:start_line]) - a:start_char + 1
   let l:line = a:start_line + 1
   while l:line < l:adj_end_line
-    let l:result += strlen(a:lines[l:line]) + 1
+    let l:result += s:count_utf16_code_units(a:lines[l:line]) + 1
     let l:line += 1
   endwhile
   let l:result += l:adj_end_char + 1
   return l:result
 endfunction
+
+function! s:count_utf16_code_units(str) abort
```

Comment:
Does this behave differently from `strchars(str)`?

Reply:
In LSP spec, `𐐀` is two characters. But strchars return 1. See https://github.com/Microsoft/language-server-protocol/issues/376